### ISSUE TITLE
`transmission-qt` crashes with a segfault after adding a magnet URI.

### ISSUE THREAD

**Issue by maxz** [transmission/transmission#2979]
The version of Transmission is `3.00 (180586d1ba)`.

This seems to be a front-end issue. The running `transmission-daemon` is not affected.
I use `transmission-qt` connected to the `transmission-daemon` via localhost.

It happened a few times now with different magnet URIs. At first I thought it only happened with very long magnet links (e. g. which contain a lot of extra tracker information after the hash), but this is not the case. It also happens with a magnet link which contains only the `exact topic` with the hash and nothing else (e.g. `magnet:?xt=urn:btih:` followed by 40 hexadecimal characters).

Nothing unusual happens if I add the same links through the RPC interface (e. g. there is nothing unusual I can observe in the response).

Even when `transmission-qt` segfaults, the torrent is added successfully. If I try to add it after the crash, I get the message that it is a duplicate and there is no further crash happening.

So the fault seems to be somewhere in the evaluation of the response to adding the torrent for the first time.
I would rather not supply the latest magnet link which caused this here and do not remember the older ones.

I do not have a lot of time right now to look at this myself, so I just wanted to report it to get it out of my mind for now.

Part of my system log in case it is any help:
```
[Apr24 06:38] traps: transmission-qt[1559] general protection fault ip:71fa0fb85df4 sp:7ffd33f7dd28 error:0 in libQt5Core.so.5.15.3[71fa0fac6000+2c9000]
[Apr24 06:42] transmission-qt[20182]: segfault at 25 ip 000072149b89adfc sp 00007fffe3ca3718 error 4 in libQt5Core.so.5.15.3[72149b7db000+2c9000]
[  +0.000020] Code: 6c 24 20 e9 73 ff ff ff e8 b1 06 f4 ff f3 0f 1e fa 49 89 c4 e9 ba 7e f4 ff 0f 1f 44 00 00 f3 0f 1e fa 48 8b 3f 48 8b 36 31 c0 <48> 63 57 04 3b 56 04 74 0b c3 66 2e 0f 1f 84 00 00 00 00 00 48 8d
```
The pointers probably will not be of much help.

**Comment by Coeur** [comment]
According to https://github.com/transmission/transmission/issues/88#issuecomment-846588490, "180586d1ba" is about one year old (24 May 2021) or older.

Support will only be given for nightly builds or nightly tarballs or builds from master branch:
https://transmissionbt.com/download/

Or you'd need to give an exact reproduction step (like a specific magnet URL), together with informations on your platform (macos, linux, windows, ...)

**Comment by ckerr** [comment]
@maxz, :+1: on what @Coeur said -- support will be given for issues in 3.00 that persist into the nightly builds, but the next step is to see if the problem persists in the nighlies. :bowing_man:

**Comment by maxz** [comment]
I tested it against commit `026174ae51fb5dfc5c2b41a69d984ca4c6ac3135` and it did not happend.
Then I tried using the 3.0+ `transmission-qt` while running the old `transmission-daemon` and it lead to a message about an invalid torrent but seems to be due to some incompatibility in the RPC interface.
I then for good measure tried adding it again with pure 3.0 (both client and server) and it did not cause a segmentation fault either, although at the time of opening the issue it happened reproducibly. So I will keep and eye on this and go through all the steps again if I run across another magnet URI which causes a crash.

I will close this issue for now and open a new one when that day comes.
